lib/extract/dl_esm_inf/netcdf/kernel_data_netcdf.f90

```diff
@@ -0,0 +1,448 @@
+    end subroutine DeclareScalarReal
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine writes the value of a scalar single precision variable
+    !! to the NetCDF file. It takes the variable id from the corresponding
+    !! declaration.
+    !! @param[inout] this The instance of the PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine WriteScalarReal(this, name, value)
+        use netcdf
+        implicit none
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        real, intent(in) :: value
+        integer :: retval
+        retval = CheckError(nf90_put_var(this%ncid, this%var_id(this%next_var_index),  &
+                                         value))
+        this%next_var_index = this%next_var_index + 1
+    end subroutine WriteScalarReal
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine reads the value of a scalar single precision variable
+    !! from the NetCDF file and returns it to the user
+    !! @param[inout] this The instance of the PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[out] value The read value is stored here.
+    subroutine ReadScalarReal(this, name, value)
+        use netcdf
+        implicit none
+
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        real :: value
+        integer :: retval, varid
+        retval = CheckError(nf90_inq_varid(this%ncid, name, varid))
+        retval = CheckError(nf90_get_var(this%ncid, varid, value))
+    end subroutine ReadScalarReal
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine writes the value of a scalar double precision variable
+    !! to the NetCDF file. It takes the variable id from the corresponding
+    !! declaration.
+    !! @param[inout] this The instance of the PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine WriteScalarDouble(this, name, value)
+        implicit none
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        double precision, intent(in) :: value
+    end subroutine WriteScalarDouble
+    
+    ! -------------------------------------------------------------------------
+    !> This subroutine declares a double precision field as defined in
+    !! dl_esm_info (r2d_field). A corresponding variable definition is added
+    !! to the NetCDF file, and the variable id is stored in the var_id field.
+    !! @param[inout] this The instance of the PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    !! @param[inout] this The instance of the PSyDataType.
```

Comment:
You've duplicated the documentation of PSyDataType but missed out that for `prefix` :-)

Reply:
Aeh ... the problem is actually that the prefix variables are unused (they were my original idea, and I forgot to remove this).